### PR TITLE
azcopy: init at 10.0.1

### PR DESCRIPTION
**Pull request by colemickens**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by colemickens** [comment]
(Note, in general, right now `azcopy` is probably taken to mean the .NET Core release, but it looks like Azure is moving to the Golang version in the future. Plus it's easier to package for now. The binary built by the repo for this package is `azure-storage-azcopy`, hence why I've named it such here.)

File: pkgs/development/tools/azcopy/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# with import <nixpkgs>{};
```
Comment (srhb) [inline pkgs/development/tools/azcopy/default.nix]: Remove this comment

File: pkgs/development/tools/azcopy/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{ stdenv, fetchFromGitHub, buildGoPackage }:
+
+buildGoPackage rec {
+  name = "azure-storage-azcopy-${version}";
+  version = "10.0.1";
```
Comment (srhb) [inline pkgs/development/tools/azcopy/default.nix]: I this this should have `pre` in it, since they are listed as prereleases on GitHub.

**Comment by colemickens** [comment]
Thanks for the review @srhb! I've made the requested changes.

**Pull request merged**